api/provider/aws/common.go

```diff
@@ -151,6 +153,51 @@ func createSG(ec2api ec2iface.EC2API, groupName, description, vpcID string) erro
 	return nil
 }
 
+func readASG(autoscalingapi autoscalingiface.AutoScalingAPI, autoScalingGroupName string) (*autoscaling.Group, error) {
+	input := &autoscaling.DescribeAutoScalingGroupsInput{}
+	input.SetAutoScalingGroupNames([]*string{aws.String(autoScalingGroupName)})
+
+	output, err := autoscalingapi.DescribeAutoScalingGroups(input)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, asg := range output.AutoScalingGroups {
+		if aws.StringValue(asg.AutoScalingGroupName) == autoScalingGroupName {
+			return asg, nil
+		}
+	}
+
+	message := fmt.Sprintf("AutoScalingGroup '%s' does not exist", autoScalingGroupName)
+	return nil, awserr.New("DoesNotExist", message, nil)
+}
+
+func updateASG(autoscalingapi autoscalingiface.AutoScalingAPI, autoScalingGroupName string, minSize, maxSize, desiredCapacity *int64) error {
```

Comment:
Why using int64 pointer when your usage requires dereferencing it anyway? Switching to a plain int64 seems like the right thing to do here.

Reply:
@sesh-kebab In moving `updateASG()` into `common.go`, I need to maintain two different workflows.
1. `EnvironmentProvider` will call this function to set min and max scale, but not desired scale
2. `EnvironmentScaler` will call this function to set desired scale, but neither min nor max scale

Using a pointer allows these params to optionally be `nil`, since a default `int64` value of `0` is actually a valid value to be pasing.

Reply:
@tlake oh i see - I didn't see that it was being called from the provider also. 👍

Reply:
:) ✅